code/modules/food_and_drinks/food/snacks_spaghetti.dm

```diff
@@ -5,101 +5,99 @@
 	icon = 'icons/obj/food/pizzaspaghetti.dmi'
 	icon_state = "spaghetti"
 	list_reagents = list("nutriment" = 1, "vitamin" = 1)
-	cooked_type = /obj/item/reagent_containers/food/snacks/boiledspaghetti
+	cooked_type = /obj/item/reagent_containers/food/snacks/spaghetti/boiledspaghetti
 	filling_color = "#F0E68C"
 	tastes = list("pasta" = 1)
 	foodtype = GRAIN
 
-/obj/item/reagent_containers/food/snacks/boiledspaghetti
+/obj/item/reagent_containers/food/snacks/spaghetti/Initialize()
+	. = ..()
+	if(!cooked_type) // This isn't cooked, why would you put uncooked spaghetti in your pocket?
+		AddComponent(/datum/component/spill, list("Something wet falls out of their pocket and hits the ground. Is that... spaghetti?", "Oh shit! All your pocket spaghetti fell out!"), 'sound/effects/splat.ogg')
```

Comment:
`suggestion
 		AddComponent(/datum/component/spill, list("Something wet falls out of their pocket and hits the ground. Is that... [src]?", "Oh shit! All your pocket [src] fell out!"), 'sound/effects/splat.ogg') `

Reply:
I didn't do that because components can be transferred. I'm overthinking it I suppose.